### PR TITLE
Extract a base connection class to for service connection and utilities

### PR DESCRIPTION
**Pull request by vicancy**
1. For AspNet support, the underlying server-service connection is also leveraging the aspnetcore http transport logic. So a base connection class is extracted for feature sharing, including:
  a. KeepAlive
  b. Process handshake request and response
  c. Process and dispatch incoming messages
2. Move utility classes here, so that it can be shared by AspNet SDK
3. The newly added Common project is Marked as `IsPackable = false` so that it will not get published, instead, it will be copied to the sdk package when `dotnet pack`. (Is it meaningful to netcore projects?)

### CODE REVIEW

File: src/Microsoft.Azure.SignalR.Common/Microsoft.Azure.SignalR.Common.csproj [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  </PropertyGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\Microsoft.Azure.SignalR.Protocols\Microsoft.Azure.SignalR.Protocols.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="System.IdentityModel.Tokens.Jwt" Version="$(SystemIdentityModelTokensJwtPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Http.Connections.Client" Version="$(MicrosoftAspNetCoreHttpConnectionsClientPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.SignalR" Version="$(MicrosoftAspNetCoreSignalRPackageVersion)" />
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR.Common/Microsoft.Azure.SignalR.Common.csproj]: This can't be a dependency in the shared code.

Comment (vicancy) [reply src/Microsoft.Azure.SignalR.Common/Microsoft.Azure.SignalR.Common.csproj]: oops

File: src/Microsoft.Azure.SignalR.Common/StaticRandom.cs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 
 namespace Microsoft.Azure.SignalR
 {
-    internal class StaticRandom
+    public class StaticRandom
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR.Common/StaticRandom.cs]: I don't like that this is public.

File: src/Microsoft.Azure.SignalR.Common/TimerAwaitable.cs [hunk 3]
```diff
@@ -1,109 +1,109 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.SignalR
+{
+    public class TimerAwaitable : IDisposable, ICriticalNotifyCompletion
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR.Common/TimerAwaitable.cs]: I don't think this should be public. Lets us InternalsVisibleTo

File: src/Microsoft.Azure.SignalR.Common/ServiceConnectionBase.cs [hunk 4]
```diff
@@ -0,0 +1,641 @@
...
+
+        protected abstract Task<ConnectionContext> CreateConnection();
+
+        protected abstract Task DisposeConnection();
+
+        protected abstract Task CleanupConnections();
+
+        protected abstract Task OnConnectedAsync(OpenConnectionMessage openConnectionMessage);
+
+        protected abstract Task PerformDisconnectAsync(CloseConnectionMessage closeConnectionMessage);
```
Comment (davidfowl) [inline src/Microsoft.Azure.SignalR.Common/ServiceConnectionBase.cs]: nit: Rename to `OnDisconnectedAsync`

**Pull request merged**